doc/source/changes/version_0_29.rst.inc

```diff
@@ -29,6 +29,32 @@ New features
     >>> a02.equals(a['a1:a3'] >> 'group_a')
     False
 
+* allowed arrays to have metadata (e.g. title, description, authors, ...).
+  Metadata can be accesed using the syntax ``array.meta.name`` or ``array.meta[name]``.
+  Warning: currently, only saving and loading to/from HDF file support metadata.
+  See below for use cases:
+
+    >>> # add metadata to/from an array
+    >>> arr = ndtest((3, 3), meta=OrderedDict([('title', 'array for testing'), ('author', 'John Smith')])
+
+    >>> # access an item
+    >>> arr.meta.author
+    'John Smith'
+    >>> # or
+    >>> arr.meta['author']
+    'John Smith'
+
+    >>> # update/add an item (the item is added if not found in existing metadata)
+    >>> arr.meta.city = 'London'
+    >>> # or
+    >>> arr.meta['city'] = 'London'
+
+    >>> # delete an item
+    >>> del arr.meta.city
+    >>> # or
+    >>> del arr.meta['city']
```

Comment:
1. More important than showing the two different syntaxes for modifying/deleting metadata would be to show metadata propagation and explain the rule used to decide whether a method propagates metadata or not. To avoid surprises, it is **extremely** important that we have a simple rule for this and that users understand it. I think the rule should be something like: all operations which only modify the axes (including reordering, subset, etc.) preserve metadata, all operations which modify the values drop metadata. I already see one notable exception: `__setitem__` should not drop metadata since we modify the object in-place.
2. All functions which take a meta argument should have three options: keep previous metadata, take some new (explicit) metadata, or drop metadata. I am still unsure which should be the default. Maybe dropping by default with an easy option to keep it might be saner and an easier rule for users to remember (FWIW, this is what xarray advertise it does, but I don't see the option in their API documentation).

Reply:
I am unsure about those but I would rather keep metadata.

Reply:
much better